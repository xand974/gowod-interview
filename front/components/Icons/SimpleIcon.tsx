import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { Icon } from "react-native-elements/dist/icons/Icon";

export type CtmIconHost =
  | "material"
  | "material-community"
  | "simple-line-icon"
  | "zocial"
  | "font-awesome"
  | "octicon"
  | "ionicon"
  | "foundation"
  | "evilicon"
  | "entypo"
  | "antdesign"
  | "font-awesome-5"
  | "ant-design";

type SimpleIconType = {
  type: CtmIconHost;
  name: string;
  size?: number;
  color?: string;
};
export default function SimpleIcon({
  type,
  name,
  size,
  color,
}: SimpleIconType) {
  return (
    <Icon
      type={type}
      name={name}
      size={size ?? 20}
      color={color ?? "#d1d1d1"}
    />
  );
}

const styles = StyleSheet.create({});
