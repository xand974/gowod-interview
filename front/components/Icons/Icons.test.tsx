import renderer from "react-test-renderer";
import SimpleIcon from "../../components/Icons/SimpleIcon";

test("render SimpleIcon component", () => {
  const component = renderer.create(
    <SimpleIcon
      type="ionicon"
      name="information-circle-outline"
      size={20}
      color="red"
    />
  );
  const componentJSON = component.toJSON() as any;
  expect(componentJSON).toMatchSnapshot();
});
