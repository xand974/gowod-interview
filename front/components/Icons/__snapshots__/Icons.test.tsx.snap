// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`render SimpleIcon component 1`] = `
<View
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
